lib/eco/treemanager.py

```diff
@@ -628,7 +629,14 @@ def jump_to_error(self, parser):
 
     def key_shift_ctrl_z(self):
         self.log_input("key_shift_ctrl_z")
-        if self.get_max_version() > self.version:
+        try:
+            i = self.undo_snapshots.index(self.version)
+            if i == len(self.undo_snapshots) - 1:
+                return
+            undo_amount = self.undo_snapshots[i+1] - self.undo_snapshots[i]
+        except ValueError:
```

Comment:
I could maybe move parts in the finally block to make clear what throws the error, e.g.:
```
        try:
            i = self.undo_snapshots.index(self.version)
        except ValueError:
            ...
        finally:
            if i == len(self.undo_snapshots) - 1:
                return
            undo_amount = self.undo_snapshots[i+1] - self.undo_snapshots[i]
```